### PR TITLE
chore: bump celestia-app to v1.0.0-rc8

### PR DESCRIPTION
**Pull request by evan-forbes**
## Overview

this PR bumps node to rc8 

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -7,10 +7,10 @@ replace github.com/ipfs/go-verifcid => github.com/celestiaorg/go-verifcid v0.0.1
 require (
 	cosmossdk.io/errors v1.0.0-beta.7
 	cosmossdk.io/math v1.0.0-rc.0
-	github.com/BurntSushi/toml v1.3.0
+	github.com/BurntSushi/toml v1.3.2
 	github.com/alecthomas/jsonschema v0.0.0-20200530073317-71f438968921
 	github.com/benbjohnson/clock v1.3.5
-	github.com/celestiaorg/celestia-app v1.0.0-rc5
+	github.com/celestiaorg/celestia-app v1.0.0-rc7.0.20230706110617-0f1ce2ce52e7
```
Comment (evan-forbes) [inline go.mod:13]: we still need to swap to the official tag once release is cut

File: core/eds.go [hunk 2]
```diff
@@ -19,22 +17,12 @@ import (
...
-	if len(data.Txs) == 0 && data.SquareSize == uint64(1) {
+func extendBlock(data types.Data, appVersion uint64) (*rsmt2d.ExtendedDataSquare, error) {
+	if app.IsEmptyBlock(data, appVersion) {
 		return nil, nil
 	}
 
-	sqr, err := square.Construct(data.Txs.ToSliceOfBytes(), appconsts.LatestVersion, share.MaxSquareSize)
-	if err != nil {
-		return nil, err
-	}
```
Comment (evan-forbes) [inline core/eds.go:30]: unrelated to this release, fwiw, node was already constructing the square in the correct way. I was originally correct when saying that rc7 was not breaking to celestia-node (notice the usage of `share.MaxSquareSize`)

File: go.mod (hunk 1 above)
Comment (adlerjohn) [reply go.mod:13]: is this PR mergable? if it's still waiting for an app rc, make it a draft to avoid confusion :)

**Review by Wondertan — APPROVED** [review]

**Review by Wondertan — CHANGES_REQUESTED** [review]
A wait linter complains

File: go.mod (hunk 1 above)
Comment (evan-forbes) [reply go.mod]: yeah sorry we can keep it a draft, as of ~30minutes ago this was updated

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

File: core/eds_test.go [hunk 3]
```diff
@@ -32,14 +34,19 @@ func TestTrulyEmptySquare(t *testing.T) {
...
+		Txs: []types.Tx{},
 	}
 
-	eds, err := extendBlock(data)
+	eds, err := extendBlock(data, appconsts.LatestVersion)
+	require.Nil(t, eds)
+	require.NoError(t, err)
+
+	// force extend the square using an empty block and compare with the min DAH
+	eds, err = app.ExtendBlock(data, appconsts.LatestVersion)
```
Comment (renaynay) [inline core/eds_test.go:47]: should we do a comparison here?

**Pull request merged**